rpc/rpc-transport/socket/src/name.c

```diff
@@ -108,27 +108,20 @@ af_unix_client_bind(rpc_transport_t *this, struct sockaddr *sockaddr,
     path_data = dict_get_sizen(this->options, "transport.socket.bind-path");
     if (path_data) {
         char *path = data_to_str(path_data);
-
-        int count = 0;
-
-        while (path[count] != '\0') {
-            count++;
-        }
-
-        addr = (struct sockaddr_un *)sockaddr;
-
-        if (count <= 108) {
-            strcpy(addr->sun_path, path);
-            ret = bind(sock, (struct sockaddr *)addr, sockaddr_len);
-
-        } else { /* 108 = addr->sun_path length */
-
+        if (!path || path_data->len > 108) { /* 108 = addr->sun_path length */
             gf_log(this->name, GF_LOG_TRACE,
                    "bind-path not specified for unix socket, "
                    "letting connect to assign default value");
             goto err;
         }
 
+        addr = (struct sockaddr_un *)sockaddr;
+
+        strncpy(addr->sun_path, path, sizeof(addr->sun_path));
+        addr->sun_path[sizeof(addr->sun_path) - 1] = '\0';
+
+        ret = bind(sock, (struct sockaddr *)addr, sockaddr_len);
+
         if (ret == -1) {
```

Comment:
remove "goto err" inside this if block as it is anyway going to err.

Reply:
Yeah true. Will remove that.